### PR TITLE
Create ACL for Klaviyo config settings

### PR DESCRIPTION
**Pull request by snoop0x7b**
## Description

This pull request creates the ACL that's raised in this issue https://github.com/klaviyo/magento2-klaviyo/issues/160

It creates permissions for each klaviyo configuration area. 

## Manual Testing Steps

1. php bin/magento cache:clean
2. Log into the admin panel
3. Go to system -> User Roles
4. Check whether the new klaviyo roles appear (control+f klaviyo)

## Pre-Submission Checklist:

- [ ] You've updated the CHANGELOG following the steps [here](https://github.com/klaviyo/magento2-klaviyo#making-updates)
- [ ] **Internal Only** - If this is a release, please confirm the following:
  - [ ] The links in the changelog have been updated to point towards the new versions
  - [ ] The version has been incremented in the following places: module.xml and composer.json

**NOTE:** Please use the [Changelogger](https://pypi.org/project/changelogged/) cli tool to manage versioned file upgrades.

### CODE REVIEW

**Comment by cykolln** [comment]
@snoop0x7b Thanks for contributing! This won't work as expected because all sections except General use the same `resource` id `Klaviyo_Reclaim::klaviyo_reclaim`. We should separate these out to match what you have in acl.xml, this should be good to go if you: 
* update system.xml to reference the resource ids you defined in acl.xml, for example: update the `<resource>` tag for section `klaviyo_reclaim_newsletter` to be `Klaviyo_Reclaim::klaviyo_reclaim_newsletter` as defined in acl.xml

File: etc/acl.xml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        <resources>
+            <resource id="Magento_Backend::admin">
+                <resource id="Magento_Backend::stores">
+                    <resource id="Magento_Backend::stores_settings">
+                        <resource id="Magento_Config::config">
+                            <resource id="Klaviyo_Reclaim::klaviyo_reclaim_general" title="Klaviyo General" translate="title" sortOrder="150" />
+                            <resource id="Klaviyo_Reclaim::klaviyo_reclaim_newsletter" title="Klaviyo Newsletter" translate="title" sortOrder="160" />
+                            <resource id="Klaviyo_Reclaim::klaviyo_reclaim_consent_at_checkout" title="Klaviyo Consent at checkout" translate="title" sortOrder="170" />
+                            <resource id="Klaviyo_Reclaim::klaviyo_reclaim_oauth" title="Klaviyo Oauth" translate="title" sortOrder="180" />
+                            <resource id="Klaviyo_Reclaim::klaviyo_reclaim_webhook" title="Klaviyo Webhook" translate="title" sortOrder="190" />
```
Comment (cykolln) [inline etc/acl.xml:13]: Lets group these under a common "Klaviyo" section so they don't accidentally get split up and are easier to find:
```suggestion
                           <resource id="Klaviyo_Reclaim::config" title="Klaviyo" translate="title" sortOrder="80">
                                 <resource id="Klaviyo_Reclaim::klaviyo_reclaim_general" title="Klaviyo General" translate="title" sortOrder="150" />
                                 <resource id="Klaviyo_Reclaim::klaviyo_reclaim_newsletter" title="Klaviyo Newsletter" translate="title" sortOrder="160" />
                                 <resource id="Klaviyo_Reclaim::klaviyo_reclaim_consent_at_checkout" title="Klaviyo Consent at checkout" translate="title" sortOrder="170" />
                                 <resource id="Klaviyo_Reclaim::klaviyo_reclaim_oauth" title="Klaviyo Oauth" translate="title" sortOrder="180" />
                                 <resource id="Klaviyo_Reclaim::klaviyo_reclaim_webhook" title="Klaviyo Webhook" translate="title" sortOrder="190" />
                            </resource>     
```

**Review by cykolln — CHANGES_REQUESTED** [review]

**Comment by cykolln** [comment]
@snoop0x7b We are looking to release a new version of the module in December and can include these changes once the feedback above has been addressed. Thanks!

**Comment by snoop0x7b** [comment]
> @snoop0x7b We are looking to release a new version of the module in December and can include these changes once the feedback above has been addressed. Thanks!

Thanks! I'll take a look and make those changes! Sorry it took me a while to get to this.

**Comment by snoop0x7b** [comment]
Thanks for the feedback on this, I have made the requested updates!

**Comment by cykolln** [comment]
@snoop0x7b Looks good to me. Can you please update the changelog and then I can approve!

**Review by cykolln — APPROVED** [review]

**Comment by cykolln** [comment]
@snoop0x7b We want to get this in the next release so i'm going to go ahead and merge this in and we will update the changelog in a separate PR to reflect. Thanks for contributing!

**Pull request merged**